bitcoin/client.go

```diff
@@ -364,63 +364,53 @@ func (b *Client) getBlock(
 	//   1. Block hash (string, required)
 	//   2. Verbosity (bool, optional, default=false)
 	params := []interface{}{hash, true}
-	responseV1 := &blockResponseV1{}
-	if err := b.post(ctx, requestMethodGetBlock, params, responseV1); err != nil {
+	blockResponse := &blockResponse{}
+	if err := b.post(ctx, requestMethodGetBlock, params, blockResponse); err != nil {
 		return nil, fmt.Errorf("%w: error fetching block by hash %s", err, hash)
 	}
-	// Parameters:
-	//   1. Block hash (string, required)
-	//   2. Verbosity (bool, optional, default=false)
-	params = []interface{}{hash, false}
-	responseV0 := &blockResponseV0{}
-	if err := b.post(ctx, requestMethodGetBlock, params, responseV0); err != nil {
-		return nil, fmt.Errorf("%w: error fetching block by hash %s", err, hash)
-	}
-	// Decode the serialized block hex to raw bytes
-	block, err := hex.DecodeString(responseV0.Result)
-	if err != nil {
-		return nil, err
-	}
-	// Deserialize the block
-	var msgBlock AuxBlock
-	if err := msgBlock.Deserialize(bytes.NewReader(block)); err != nil {
-		return nil, err
-	}
 
-	// Decode each transaction in the block and append results to txs
-	var txs []*Transaction
-	for _, tx := range msgBlock.Transactions {
-		buf := bytes.NewBuffer(make([]byte, 0, tx.SerializeSize()))
-		if err := tx.Serialize(buf); err != nil {
+	// if Txs == 0 fetch Transactions
+	if len(blockResponse.Result.Txs) == 0 {
+		// Parameters:
+		//   1. Block hash (string, required)
+		//   2. Verbosity (bool, optional, default=false)
+		params = []interface{}{hash, false}
+		stringResponse := &stringResponse{}
+		if err := b.post(ctx, requestMethodGetBlock, params, stringResponse); err != nil {
+			return nil, fmt.Errorf("%w: error fetching block by hash %s", err, hash)
+		}
+		// Decode the serialized block hex to raw bytes
+		block, err := hex.DecodeString(stringResponse.Result)
```

Comment:
nit: can we use inline declaration and error handling where available?

ie; could be:
```
		if block, err := hex.DecodeString(stringResponse.Result); err != nil {
			return nil, err
		}
```

Reply:
Added issue #47 for this.